rq/timeouts.py

```diff
@@ -5,18 +5,31 @@
 import signal
 
 
-class JobTimeoutException(Exception):
+class BaseTimeoutException(Exception):
+    """Base exception for timeouts."""
+    pass
+
+
+class JobTimeoutException(BaseTimeoutException):
     """Raised when a job takes longer to complete than the allowed maximum
     timeout value.
     """
     pass
 
 
+class HorseMonitorTimeoutException(BaseTimeoutException):
+    """Raised when waiting for a horse exiting takes longer than the maximum
+    timeout value.
+    """
+    pass
+
+
 class BaseDeathPenalty(object):
     """Base class to setup job timeouts."""
 
-    def __init__(self, timeout):
+    def __init__(self, timeout, exception):
```

Comment:
`exception` should default to `JobTimeoutException`, this may be used by other custom workers so we should keep this backward compatible.

Reply:
Done.